### PR TITLE
Travis Support

### PR DESCRIPTION
**Pull request by danrschlosser**
Review: @natebrennand 

- Write `.travis.yml`
- Move `assert` statements to `__all__ = [ ... ]` statements
- Fix flake8 errors
- Configure all errors to `raise Exception()` as opposed to `print()`, in order to avoid nosetests consuming error output.
- Setup `nosetests` for future testing

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - MONGO_DATABASE=eventum
+    # Logging settings
+    - LOG_FILE_MAX_SIZE=256   # in MB
+    - APP_LOG_NAME=log/app.log
+    - WERKZEUG_LOG_NAME=log/werkzeug.log
+
+git:  # We need custom git configurations
+    submodules: false
+before_install:  # Use sed to replace the SSH URL with the public URL
+    - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules
```
Comment (natebrennand) [inline .travis.yml]: shouldn't the `.gitmodules` just use `https` anyways?

File: app/__init__.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def register_logger():
     """Create an error logger and attach it to ``app``."""
 
     max_bytes = int(app.config["LOG_FILE_MAX_SIZE"]) * 1024 * 1024   # MB to B
-    Handler = logging.handlers.RotatingFileHandler
+    Handler = logging.handlers.RotatingFileHandler  # noqa
```
Comment (natebrennand) [inline app/__init__.py]: noqa?

File: app/forms/__init__.py [hunk 3]
```diff
@@ -7,11 +7,9 @@
...
-assert (CreateBlogPostForm,
-        CreateEventForm,
-        CreateProfileForm,
-        DeleteEventForm,
-        EditEventForm,
-        UploadImageForm,
-        AddToWhitelistForm,
-        EditUserForm)
\ No newline at end of file
+# Silences flake8
```
Comment (natebrennand) [inline app/forms/__init__.py]: clarify this slightly more?
"Silence flake8 by referencing objects"?

File: app/forms/fields/TimeField.py [hunk 4]
```diff
@@ -20,7 +21,11 @@ class TimeField(Field):
     widget = TimeInput()
     error_msg = 'Not a valid time.'
 
-    def __init__(self, label=None, validators=None, format='%I:%M%p', **kwargs):
+    def __init__(self,
+                 label=None,
+                 validators=None,
+                 format='%I:%M%p',
```
Comment (natebrennand) [inline app/forms/fields/TimeField.py]: what is this format? can you include an example string?

File: app/forms/fields/__init__.py [hunk 5]
```diff
@@ -1,3 +1,3 @@
 from TimeField import TimeField
 
-assert TimeField
\ No newline at end of file
+__all__ = ['TimeField']
```
Comment (natebrennand) [inline app/forms/fields/__init__.py]: needs comment

File: app/forms/validators.py [hunk 6]
```diff
@@ -68,7 +72,7 @@ def __init__(self, original, message="An event with that slug already exists."):
         :param str message: An alternate message to be raised.
         """
         self.original = original
-        self.message = message
+        self.message = message if message else self.SLUG_MESSAGE
```
Comment (natebrennand) [inline app/forms/validators.py]: what's clearer?
or:

```py
self.message = message or self.SLUG_MESSAGE
```

Python ternary statements are weird IMO

File: app/lib/google_calendar_resource_builder.py [hunk 7]
```diff
@@ -65,15 +66,17 @@ def _recurrence(klass, s):
...
             r += ';INTERVAL=%d' % s.every
         if s.ends_on:
-            r += ';UNTIL={}'.format(klass._recurrence_end_date(s))
+            r += ';UNTIL={}'.format(cls._recurrence_end_date(s))
         elif s.ends_after:
             r += ';COUNT=%d' % s.num_occurrences
         else:
-            raise GoogleCalendarAPIError('series should either end on or after')
+            raise GoogleCalendarAPIError(
+                'Series.ends_on and ends_after both false'
```
Comment (natebrennand) [inline app/lib/google_calendar_resource_builder.py]: Clarify this error message more?

File: app/models/User.py [hunk 8]
```diff
@@ -117,7 +118,8 @@ def get_profile_picture(self, size=50):
         if self.image:
             return self.image.url()
         if not self.image_url:
-            return 'https://lh6.googleusercontent.com/-K9HZ5Z5vOU8/AAAAAAAAAAI/AAAAAAAAAAA/yRoMtBSXoxQ/s48-c/photo.jpg'
+            return ('https://lh6.googleusercontent.com/-K9HZ5Z5vOU8/'
```
Comment (natebrennand) [inline app/models/User.py]: is it possible to have a better URL here?
(not needed for this PR, but could be an issue for later)

File: app/models/__init__.py [hunk 9]
```diff
@@ -6,10 +6,7 @@
...
 
-assert (BlogPost,
-        Event,
-        EventSeries,
-        Image,
-        Post,
-        User,
-        Whitelist)
\ No newline at end of file
+# Silences flake8
```
Comment (natebrennand) [inline app/models/__init__.py]: better message here too

File: app/models/fields/TimeField.py [hunk 10]
```diff
@@ -72,9 +73,9 @@ def prepare_query_value(self, op, value):
...
             return value
         if isinstance(value, time):
-            return value.hour * 3600 + \
-                   value.minute * 60 + \
-                   value.second + \
-                   value.microsecond / 1000000
+            return (value.hour * 3600 +
+                    value.minute * 60 +
+                    value.second +
+                    value.microsecond / 1000000)
```
Comment (natebrennand) [inline app/models/fields/TimeField.py]: replace that with `10**6`?

File: app/routes/admin/media.py [hunk 11]
```diff
@@ -29,20 +29,23 @@ def index():
...
     """
-    images =Image.objects()
+    images = Image.objects()
     form = UploadImageForm()
     return render_template('admin/media/media.html', images=images, form=form)
 
 
-def allowed_file(filename):
-    """Returns True if ``filename`` is a valid filename.  It must have an
+def allowed_file(fn):
```
Comment (natebrennand) [inline app/routes/admin/media.py]: why change this var name? `fn` is often used to talk about functions

File: app/routes/admin/posts.py [hunk 12]
```diff
@@ -44,15 +46,19 @@ def new():
...
+        for u in User.objects()
+    ]
     form.author.data = str(g.user.id)
     upload_form = UploadImageForm()
     if form.validate_on_submit():
         author = User.objects().get(id=ObjectId(form.author.data))
         post = BlogPost(title=form.title.data,
                         slug=form.slug.data,
-                        images=[Image.objects().get(filename=fn) for fn in form.images.data],
+                        images=[
```
Comment (natebrennand) [inline app/routes/admin/posts.py]: i don't think this should be in the instantiation, it's messy.

File: app/models/User.py (hunk 8 above)
Comment (danrschlosser) [inline app/models/User.py]: https://github.com/adicu/adi-website/issues/204

Comment (natebrennand) [inline app/models/User.py]: :+1:

File: app/routes/admin/posts.py (hunk 12 above)
Comment (natebrennand) [inline app/routes/admin/posts.py]: to clarify, i think putting it before `post` is instantiated would be neater

**Comment by natebrennand** [comment]
:+1:

**Pull request merged**